web/react/components/sidebar.jsx

```diff
@@ -235,103 +247,143 @@ module.exports = React.createClass({
             }
         }
     },
+    onScroll: function(e) {
+        this.updateUnreadIndicators();
+    },
+    onResize: function(e) {
+        this.updateUnreadIndicators();
+    },
+    updateUnreadIndicators: function() {
+        var container = $(this.refs.container.getDOMNode());
+
+        if (this.firstUnreadChannel) {
```

Comment:
That's not a problem. When a channel gets changed to unread on another device, the component rerenders with both unread markers set back to display: none. updateUnreadIndicators is called afterwards to make the markers visible if they're still necessary.